WcaOnRails/spec/models/user_spec.rb

```diff
@@ -353,6 +353,27 @@
     end
   end
 
+  describe "check if registration form sends mail to newly registered user" do
+    it "send mail" do
+      new_user = FactoryBot.create :user, :unconfirmed
+      expect(ActionMailer::Base.deliveries.last.to).to eq [new_user.email]
+      expect(ActionMailer::Base.deliveries.last.subject).to eq "Confirmation instructions"
+    end
+  end
+
+  describe "let user change his mail address" do
+    it "send mail" do
+      user = FactoryBot.create :user
+      user.unconfirmed_email = "new@example.com"
+      expect(user.pending_reconfirmation?).to eq true
+      user.send_reconfirmation_instructions
+
+      expect { user.send_reconfirmation_instructions } .to change { ActionMailer::Base.deliveries.count }.by(1)
```

Comment:
Instead of explicitly calling `user.send_reconfirmation_instructions`, can you instead do a `user.save!` here? The problem with the test as currently written is that it checks that the `send_reconfirmation_instructions` method does the right thing, but it doesn't check that devise actually calles the `send_reconfirmation_instructions` method.